src/test/java/com/xceptance/xlt/api/util/XltCharBufferTest.java

```diff
@@ -1088,6 +1122,105 @@ public void equalsTest()
+        // identical, hashcode touched
+        {
+            var s1 = XltCharBuffer.valueOf("foo");
+            var s2 = XltCharBuffer.valueOf("foo");
+            // caches hash codes
+            assertEquals(s1.hashCode(), s2.hashCode());
+            assertTrue(s1.equals(s2));
+            assertTrue(s2.equals(s1));
+        }
+        // identical, hashcode not touched
+        {
+            var s1 = XltCharBuffer.valueOf("foo");
+            var s2 = XltCharBuffer.valueOf("foo");
+            assertTrue(s1.equals(s2));
+            assertTrue(s2.equals(s1));
+        }
+        // different via hashcode, hashcode touched
+        {
+            var s1 = XltCharBuffer.valueOf("foo2");
+            var s2 = XltCharBuffer.valueOf("foo3");
+            // caches hash codes
+            assertNotEquals(s1.hashCode(), s2.hashCode());
+            assertFalse(s1.equals(s2));
+            assertFalse(s2.equals(s1));
+        }
+        // different hashcode NOT touched
+        {
+            var s1 = XltCharBuffer.valueOf("foo2");
+            var s2 = XltCharBuffer.valueOf("foo3");
+            assertFalse(s1.equals(s2));
+            assertFalse(s2.equals(s1));
+        }
+
+        // same hashcode, same length, same first char, but different
+        {
+            var s1 = new XltCharBufferSameHash("foo2".toCharArray(), 0);
+            var s2 = new XltCharBufferSameHash("foo3".toCharArray(), 0);
+            // just check that the override is right
+            assertEquals(0, s1.hashCode());
+            assertEquals(0, s2.hashCode());
+            assertFalse(s1.equals(s2));
+            assertFalse(s2.equals(s1));
+        }
+        // same hashcode, same length, same first char, but different
+        {
+            var s1 = new XltCharBufferSameHash("foo2".toCharArray(), 82);
+            var s2 = new XltCharBufferSameHash("foo3".toCharArray(), 82);
+            // just check that the override is right
+            assertEquals(s1.hashCode(), s2.hashCode());
+            assertFalse(s1.equals(s2));
+            assertFalse(s2.equals(s1));
+        }
+        // one has cached hashcode, was has not
+        {
+            var s1 = new XltCharBufferSameHash("foo2".toCharArray(), 0);
+            var s2 = new XltCharBufferSameHash("foo3".toCharArray(), 1);
+            // just check that the override is right
+            assertEquals(0, s1.hashCode());
+            assertEquals(1, s2.hashCode());
+            assertFalse(s1.equals(s2));
+            assertFalse(s2.equals(s1));
+        }
+        // one has cached hashcode, was has not
+        {
+            var s1 = new XltCharBufferSameHash("foo2".toCharArray(), 1);
+            var s2 = new XltCharBufferSameHash("foo3".toCharArray(), 0);
+            // just check that the override is right
+            assertEquals(1, s1.hashCode());
+            assertEquals(0, s2.hashCode());
+            assertFalse(s1.equals(s2));
+            assertFalse(s2.equals(s1));
+        }
```

Comment:
All these tests mess with the hash code a lot, but the `equals` method is actually completely unaffected by the hash code. Remove?

Reply:
Removed